packages/router/router.ts

```diff
@@ -2362,8 +2384,7 @@ export function createRouter(init: RouterInit): Router {
     });
   }
 
-  function deleteFetcher(key: string): void {
-    let fetcher = state.fetchers.get(key);
+  function cleanupFetcher(key: string, fetcher: Fetcher | undefined): void {
```

Comment:
Extract `cleanupFetchers` as a way to clean them up without touching state (since we need this inside `updateState` calls for the delayed cleanup)